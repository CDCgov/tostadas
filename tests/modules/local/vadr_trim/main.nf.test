nextflow_process {

    name "Test Process VADR_TRIM"
    script "modules/local/vadr_trim/main.nf"
    process "VADR_TRIM"

    test("Should run without failures") {

        when {
            params {
                prefix = "NY0006"
            }
            process {
                """
                input[0] = [ [ batch_id: "batch_1",  batch_tsv: file("${projectDir}/tests/modules/local/repeatmasker_annotation/batch_1.tsv"), sample_id: "NY0006" ], 
                            file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta"), 
                            file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R1_001.fastq.gz"), 
                            file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

}
