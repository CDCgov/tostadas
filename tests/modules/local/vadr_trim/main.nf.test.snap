{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.trimmed.fasta:md5,9491e6440082e1f09fce1eb83b2f26e8"
                    ]
                ],
                "trimmed_fasta": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.trimmed.fasta:md5,9491e6440082e1f09fce1eb83b2f26e8"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:23:13.094608765"
    }
}